dao-factory/README.md

```diff
@@ -0,0 +1,237 @@
+---
+layout: pattern
+title: Dao Factory
+folder: dao-factory
+permalink: /patterns/dao-factory/
+categories: Creational
+language: en
+tags:
+- Decoupling
+---
+
+## Intent
+The Dao Factory Pattern provides an abstract DAO factory object (Abstract Factory) that can construct various types of concrete DAO factories, each factory supporting a different type of persistent storage implementation. Once you obtain the concrete DAO factory for a specific implementation, you use it to produce DAOs supported and implemented in that implementation.
+
+## Explanation
+
+Real world example
+
+> There are a few databases and if the implementation changes from Derby to another product, the only required change is the getDAOFactory() method call to the DAO factory to obtain a different factory.
+> 
```

Comment:
Can you paint the picture on a little bit higher abstraction level e.g. there is an application that is used to route nuclear missiles to their targets. The application uses Mongo database in production, but the developers like to use Derby for easy development and testing. DAO Factory pattern enables switching the used database easily.